apps/whale-api/migrate.js

```diff
@@ -0,0 +1,61 @@
+const fs = require('fs')
+const path = require('path')
+
+/*
+  This is a temporary migration script which gets run on commit to ensure that
+  path based imports are replaced with relative imports.
+
+  This migration script exists so that we can easily move updates from the
+  original whale repo which still has quite a few outstanding PRs being worked
+  on.
+
+  Once original whale no longer has outstanding PRs associated with it we can
+  delete this migration script.
+
+  (todo @kodemon)
+ */
+
+async function migrate (src, depth = 0) {
+  await migrateDirectory(src, depth)
+}
+
+async function migrateDirectory (src, depth) {
+  const dir = await fs.promises.opendir(src)
+  for await (const dirent of dir) {
+    if (dirent.isDirectory()) {
+      await migrateDirectory(path.resolve(src, dirent.name), depth + 1)
+    } else {
+      const filePath = path.resolve(src, dirent.name)
+      const file = await migrateImports(filePath, getRelativePathFromDepth(depth))
+      fs.writeFileSync(filePath, file)
+    }
+  }
+}
+
+async function migrateImports (src, relativePath) {
+  const file = fs.readFileSync(src, 'utf-8')
+  const lines = file.split('\n')
+  const refactored = []
+  for (const line of lines) {
+    refactored.push(
+      line
+        .replace(/'@(.*)\/dist\/(.*)'/, "'@$1'")
+        .replace(/'@whale-api-client(.*)'/, "'@defichain/whale-api-client'")
+        .replace('@src/', relativePath)
+    )
+  }
+  return refactored.join('\n')
+}
+
+function getRelativePathFromDepth (depth) {
+  if (depth === 0) {
+    return './'
+  }
+  return new Array(depth).fill('../').reduce((path, val) => {
+    return path + val
+  }, '')
+}
+
+migrate(path.resolve(__dirname, 'src')).then(() => {
+  process.exit(0)
+})
```

Comment:
This script updates all the import paths to no longer use alias paths defined in [whale](https://github.com/DeFiCh/whale/blob/main/tsconfig.json#L18-L21) as they are not supported in jellyfish structure. Alias paths are defined in the jellyfish root and jest does not properly support sub aliases.

My solution to this is to just make all the `@src` imports relative and remove any `/dist` import targets to be pure package imports. Eg. `jellyfish-package/dist/...` => `jellyfish-package`. Because of this I have updated a couple of jellyfish package [exports](https://github.com/DeFiCh/jellyfish/blob/kodemon/whale-api-client-migration/packages/jellyfish-api-core/src/index.ts#L38-L55) to support this.

Once we run migration there is only a handful of imports that needs updating. This approach allows us to simply just copy the entire `src` folder from whale to jellyfish. Run this script, update a couple of imports and we are good to go. Once we are ready to switch our development efforts of whale in jellyfish ecosystem we can run a final migration pass and delete this file.